src/containers/AppContainer.js

```diff
@@ -38,19 +38,57 @@ class AppContainer extends Component {
     var savedVfrCharts = getSavedCharts().map(function(chart) {
       return new VFRChart(chart);
     });
-
     this.setState({
       savedVfrCharts: savedVfrCharts,
       vfrChartsToShow: savedVfrCharts
     });
   }
+  //handle just like menu press so update both savedVfrCharts and vfrChartsToShow
+  handleFavPress(chartId: number){
+    let savedVfrCharts = [];
+    let vfrChartsToShow = [];
+    let vfrChart;
+
+    for(let i = 0; i < this.state.savedVfrCharts.length; i++){
+      savedVfrCharts[i] = this.state.savedVfrCharts[i];
+      if(savedVfrCharts[i].uniqueId == chartId){
+        savedVfrCharts[i].isFavorited = !savedVfrCharts[i].isFavorited;
+      }
+    }
+
+    switch(this.state.route){
```

Comment:
gonna hop on this one tomorrow but yea that makes sense i remember in the article it saying that its not a state if it can be calculated by other states or variables.